File: fta-backend/builder/src/main/java/com/kerfaiyassine/builder/services/BuilderStadiumService.java
package com.kerfaiyassine.builder.services;

import com.kerfaiyassine.builder.DTOs.OperationRequestDTO;
import com.kerfaiyassine.builder.DTOs.StadiumRequestDTO;
import com.kerfaiyassine.builder.DTOs.StadiumStatsDTO;
import com.kerfaiyassine.builder.clients.StadiumClient;

import org.springframework.stereotype.Service;

@Service
public class BuilderStadiumService {

    private final StadiumClient stadiumClient;

    public BuilderStadiumService(StadiumClient stadiumClient) {
        this.stadiumClient = stadiumClient;
    }

    public Object addStadium(StadiumRequestDTO dto) {
        return stadiumClient.addStadium(dto);
    }

    public Object getBuilderStadiums(Integer builderId) {
        return stadiumClient.getBuilderStadiums(builderId);
    }

    public StadiumStatsDTO getStats() {
        return stadiumClient.getStats();
    }

    public Object getOperations(String stadiumId) {
        return stadiumClient.getOperations(stadiumId);
    }

    public Object applyOperation(
            String stadiumId,
            OperationRequestDTO dto
    ) {
        return stadiumClient.addOperation(stadiumId, dto);
    }
}

File: fta-ui/src/app/Shared/services/builder-stadium/builder-stadium.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class BuilderStadiumService {

  private api =
    'http://localhost:8053/api/builders/stadiums';

  constructor(
    private http: HttpClient
  ) {}

  addStadium(dto: any) {

    return this.http.post(
      this.api,
      dto
    );
  }

  getBuilderStadiums(
    builderId: number
  ) {

    return this.http.get(
      `${this.api}/builder/${builderId}`
    );
  }

  getStats() {

    return this.http.get(
      `${this.api}/stats`
    );
  }

  getOperations(
    stadiumId: string
  ) {

    return this.http.get(
      `${this.api}/${stadiumId}/operations`
    );
  }

  getPendingOperations(
    stadiumId: string
  ) {

    return this.http.get(
      `${this.api}/${stadiumId}/operations/pending`
    );
  }

  getCompletedOperations(
    stadiumId: string
  ) {

    return this.http.get(
      `${this.api}/${stadiumId}/operations/completed`
    );
  }

  getBuilderOperations(
    builderId: number
  ) {

    return this.http.get(
      `${this.api}/builder/${builderId}/operations`
    );
  }

  getBuilderPendingOperations(
    builderId: number
  ) {

    return this.http.get(
      `${this.api}/builder/${builderId}/operations/pending`
    );
  }

  getBuilderCompletedOperations(
    builderId: number
  ) {

    return this.http.get(
      `${this.api}/builder/${builderId}/operations/completed`
    );
  }

  applyOperation(
    stadiumId: string,
    dto: any
  ) {

    return this.http.post(
      `${this.api}/${stadiumId}/operations`,
      dto
    );
  }

}
